### DISCUSSION TITLE
Tag co-op support for osu!taiko

### DISCUSSION THREAD

**Discussion by overdahedge2014** [ppy/osu#24759] [Ideas]
Currently on osu!stable, multiplayer tag co-op is supported and intended for osu! but not for osu!taiko/mania/catch, but with New Combo (NC) placements in the editor and the '!mp start' command, taiko beatmaps can be played this way. Over the years a few taiko players have noticed this and added NCs to their maps with this in mind, but it still isn't widely known, and I've found it enjoyable enough to play that I'd like to see it as a fully supported feature in osu!lazer.

[POV of how this currently works on stable](https://meeee.s-ul.eu/TPDR6eZq) (can skip to 0:30; normal-looking notes are for me to hit, grey notes are normal-looking to the other player and are for them to hit)
[Tag taiko info-dump Google Doc](https://docs.google.com/document/d/1v1fuYRbGFgLN2dC0fTekqRvjYYag9PJkADw9ccp7cvI/edit?usp=sharing) (this isn't necessary to read through, but it does include many more specific details I've noticed while playing tag)

The main barrier to _playing_ tag co-op on stable is the lack of existing compatible maps, since the majority of taiko mappers who don't know/care about this have no real reason to place NCs, but with more people learning about this recently there are a decent amount of maps for it now, many of which can be found with [this search](https://osu.ppy.sh/beatmapsets?m=1&q=tag%20co-op&s=any) or [this search,](https://osu.ppy.sh/beatmapsets?m=1&q=%22tag%20co-op%22&s=any) or in [this Drive folder.](https://drive.google.com/drive/folders/1eTp2TZZq5ZSWZLr6iMN0MbTwG8jDzXTE?usp=sharing)

Some kind of 'co-op player change' marker tied to a hitobject (similar to what the current NC system does) sounds fine to me as a way to handle this in editor, but the taiko mapping community might have their own ideas or suggestions for this, so if this is implemented it's probably worth asking them how they'd like to see it.

**Comment by mikumin27** [comment]
holy this is sick
